hazelcast/src/test/java/com/hazelcast/map/impl/nearcache/NearCacheTest.java

```diff
@@ -229,6 +229,111 @@ public void testNearCacheEviction_withMapClear() {
         });
     }
 
+    @Test
+    public void testNearCacheStatsWithReadOnlyProcessor() {
+        int mapSize = 1000;
+        String mapName = randomMapName();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(newNearCacheConfig().setInvalidateOnChange(false));
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance[] instances = factory.newInstances(config);
+
+        // populate map
+        IMap<Integer, Integer> map = instances[0].getMap(mapName);
+        populateMap(map, mapSize);
+
+        // populate Near Cache
+        populateNearCache(map, mapSize);
+
+        NearCacheStats stats = getNearCacheStats(map);
+
+        // make some hits
+        populateNearCache(map, mapSize);
+        long invalidationsBefore = stats.getInvalidations();
+        for (int i = 0; i < mapSize; i++) {
+            map.executeOnKey(i, new TestReadOnlyProcessor());
+        }
+        long invalidationsAfter = stats.getInvalidations();
+        assertEquals("No Invalidation after getting keys from read only entry processor ", invalidationsBefore, invalidationsAfter);
+    }
+
+
+    @Test
+    public void testNearCacheStatsWithReadOnlyProcessor_submitToKey() {
+        int mapSize = 1000;
+        String mapName = randomMapName();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(newNearCacheConfig().setInvalidateOnChange(false));
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance[] instances = factory.newInstances(config);
+
+        // populate map
+        IMap<Integer, Integer> map = instances[0].getMap(mapName);
+        populateMap(map, mapSize);
+
+        // populate Near Cache
+        populateNearCache(map, mapSize);
+
+        NearCacheStats stats = getNearCacheStats(map);
+
+        // make some hits
+        populateNearCache(map, mapSize);
+
+        long hitsBefore = stats.getHits();
```

Comment:
If we are not using hitsBefore and hitsAfter we could remove them and the assertion that is commented out

Reply:
Removed